dataRetriever/factory/requestersContainer/shardRequestersContainerFactory.go

```diff
@@ -136,6 +141,18 @@ func (srcf *shardRequestersContainerFactory) generateMetablockHeaderRequesters()
 	return srcf.container.Add(identifierHdr, requester)
 }
 
+func (srcf *shardRequestersContainerFactory) generateReceiptRequesters() error {
+	// only one receipts topic
+	selfShardID := srcf.shardCoordinator.SelfId()
+	identifierReceipt := factory.ReceiptTopic
```

Comment:
```suggestion
	identifierReceipt := factory.ReceiptTopic + srcf.shardCoordinator.CommunicationIdentifier(selfShardID)
```
in order to make it intra shard